Prog/Cvd19_affctd_ind_pop.sas

```diff
@@ -0,0 +1,254 @@
+run;
+
+proc means data=A n sum mean min max;
+  var inc: cvd19_affctd_inc: ;
+run;
+
+proc summary data=A;
+  by year serial;
+  var cvd19_affctd_ind inc: cvd19_affctd_inc: ;
+  output out=cvd19_affctd_ind_hh (drop=_type_ _freq_) sum= /autoname;
+run;
+
+
+data Covid19.cvd19_affctd_ind_pop;
+
+  merge A cvd19_affctd_ind_hh;
+  by year serial;
+  
+  ** Race/ethnicity **;
+  
+  if hispand = 0 then do;
+    if raced=100 then race_ethn = 1;  /** White non-Hispanic **/
+    else if raced=200 then race_ethn = 2;  /** Black non-Hispanic **/
+    else if 400 <= raced <= 679 then race_ethn = 4;  /** Asian non-Hispanic **/
+    else race_ethn = 5;  /** All other non-Hispanic **/
+  end;
+  else race_ethn = 3;  /** Hispanic **/
+  
+  ** Income changes **;
+  
+  if cvd19_affctd_incearn > 0 then inc_less_cvd19_affctd = inctot - cvd19_affctd_incearn;
+  if cvd19_affctd_incearn_sum > 0 then inc_less_cvd19_affctd_sum = inctot_sum - cvd19_affctd_incearn_sum;
+  
+  if inctot_sum > 0 then 
+    pct_inc_less_cvd19_affctd_sum = 100 * ( 1 - ( inc_less_cvd19_affctd_sum / inctot_sum ) );
+  
+  ** Housing cost ratio **;
+  
+  %macro hsg_cost_ratio( inc=, var= );
+  
+    if &inc > 0 then do;
+      if ownershp = 2 then &var = ( rentgrs * 12 ) / &inc;
+      else if ownershp = 1 then &var = ( owncost * 12 ) / &inc;
+      else if ownershp = 0 then &var = .n;
+    end;
+    else do;
+      if ownershp = 2 and rentgrs > 0 then &var = 1;
+      else if ownershp = 1 and owncost > 0 then &var = 1;
+      else if ownershp = 0 then &var = .n;
+      else &var = 0;
+    end;
+  
+  %mend hsg_cost_ratio;
+  
+  %hsg_cost_ratio( inc=inctot_sum, var=hsg_cost_ratio )
+  
+  %hsg_cost_ratio( inc=inc_less_cvd19_affctd_sum, var=hsg_cost_ratio_cvd19 )
+  
+  ** HUD income catagories **;
+  
+  hud_inc_orig = hud_inc;
+  
+  if not( missing( inc_less_cvd19_affctd_sum ) ) then do;
+    %Hud_inc_all( hhinc=inc_less_cvd19_affctd_sum, hhsize=numprec )
+  end;
+  
+  format hud_inc_orig Hud_inc hudinc.;
+
+  rename hud_inc_orig=hud_inc hud_inc=hud_inc_cvd19;
+
+  format cvd19_affctd_ind_Sum ;
+  
+run;
+
+%File_info( data=Covid19.cvd19_affctd_ind_pop, freqvars=mwcog_region cvd19_affctd_ind race_ethn fulltime yearround )
+
+proc freq data=Covid19.cvd19_affctd_ind_pop;
+  tables cvd19_affctd_ind * empstatd * fulltime * yearround /list missing nopercent nocum;
+run;
+
+proc univariate data=Covid19.cvd19_affctd_ind_pop;
+  var hsg_cost_ratio hsg_cost_ratio_cvd19;
+run;
+
+** Tables **;
+
+%Fmt_ind_2017f()
+
+title3 'Workers in COVID-19 affected industries by industry';
+
+proc tabulate data=Covid19.cvd19_affctd_ind_pop format=comma16.0 noseps missing;
+  where cvd19_affctd_ind = 1;
+  weight perwt;
+  class ind;
+  var total inctot incearn cvd19_affctd_incearn;
+  table 
+
+    /** Rows **/
+    all='Washington metro area (excl WV)' ind=' ',
+
+    /** Columns **/
+    sum='Workers' * total=' ' * f=comma12.0
+
+    sum='Annual income ($ 2018), 2014-18' * 
```

Comment:
Again, probably not worth doing but it would be helpful to make a macro variable for ACS year and inflation-year if we are going to run this again in the future.